spec/ruby/core/string/lstrip_spec.rb

```diff
@@ -50,4 +50,10 @@
     -> { "hello".freeze.lstrip! }.should raise_error(FrozenError)
     -> { "".freeze.lstrip!      }.should raise_error(FrozenError)
   end
+
+  it "raises an ArgumentError if the first codepoint is invalid" do
+    s = "\xDFabc".force_encoding(Encoding::UTF_8)
+    s.valid_encoding?.should be_false
+    -> { s.send(@method) { } }.should raise_error(ArgumentError)
```

Comment:
There is no `@method` here, I guess it should be `s.lstrip!`.
I think it's also good to check the message if we match CRuby, as that would notably catch this accidentally-passing spec.

Reply:
Bah, you're right. I had it in the shared spec file originally, but I needed the invalid code point to be at the front of the string for `lstrip` and the back of the string for `rstrip`. I'll fix.

Reply:
I was curious how these were passing then. It turns out that `@method` still has a valid value from earlier executions. I'm changing it regardless, but that was surprising. Maybe `it_behaves_like` should clean up its ivars after it's done.

Reply:
Right, I'll look into that in MSpec. I assumed `send(nil)` was ArgumentError but it's actually TypeError.

Reply:
https://github.com/ruby/mspec/commit/283710cc9128a9efe25d3db14f4828d73b97dd94